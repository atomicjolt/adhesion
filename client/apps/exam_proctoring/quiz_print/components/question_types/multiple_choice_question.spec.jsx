import React            from 'react';
import { shallow } from 'enzyme';
import _                from 'lodash';
import MultipleChoice   from './multiple_choice_question';

describe('Multiple Choice', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      answers: [
        {
          id: 1,
          text: 'yes'
        },
        {
          id: 2,
          text: 'no'
        },
        {
          id: 3,
          text: 'maybe'
        },
      ],
      question_text: '<p>I am the title</p>',
      id: 7,
    };
    result = shallow(<MultipleChoice {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders radio buttons for the options', () => {
    const radios = result.find('input');
    expect(radios.length).toBe(3);
    expect(radios.at(0).props().type).toEqual('radio');
  });
});
