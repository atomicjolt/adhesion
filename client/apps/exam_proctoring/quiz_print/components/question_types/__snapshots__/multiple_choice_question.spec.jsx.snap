// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Choice matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <multipleChoiceQuestion
    answers={
        Array [
            Object {
              "id": 1,
              "text": "yes",
            },
            Object {
              "id": 2,
              "text": "no",
            },
            Object {
              "id": 3,
              "text": "maybe",
            },
          ]
    }
    id={7}
    question_text="<p>I am the title</p>"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <essayQuestion
          text="<p>I am the title</p>"
/>,
        Array [
          <div>
            <hr />
            <input
                        disabled={true}
                        type="radio"
            />
             
            yes
</div>,
          <div>
            <hr />
            <input
                        disabled={true}
                        type="radio"
            />
             
            no
</div>,
          <div>
            <hr />
            <input
                        disabled={true}
                        type="radio"
            />
             
            maybe
</div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "<p>I am the title</p>",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "question7_answer1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="radio"
/>,
            " ",
            "yes",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "yes",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "question7_answer2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="radio"
/>,
            " ",
            "no",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "no",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "question7_answer3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="radio"
/>,
            " ",
            "maybe",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "radio",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "maybe",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <essayQuestion
            text="<p>I am the title</p>"
/>,
          Array [
            <div>
              <hr />
              <input
                            disabled={true}
                            type="radio"
              />
               
              yes
</div>,
            <div>
              <hr />
              <input
                            disabled={true}
                            type="radio"
              />
               
              no
</div>,
            <div>
              <hr />
              <input
                            disabled={true}
                            type="radio"
              />
               
              maybe
</div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "<p>I am the title</p>",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "question7_answer1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="radio"
/>,
              " ",
              "yes",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "yes",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "question7_answer2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="radio"
/>,
              " ",
              "no",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "no",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "question7_answer3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="radio"
/>,
              " ",
              "maybe",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "radio",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "maybe",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
